File: Bit_Manipulation/268_missing_number/268.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int result = nums.size();
        for (int i = 0; i < nums.size(); i++) {
            result ^= (i ^ nums[i]);
        }
        return result;
    }
};

File: Bit_Manipulation/268_missing_number/268.rs


impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let mut result: i32 = nums.len() as i32;
        let mut i: i32 = 0;
        for num in nums.iter() {
            result ^= (i ^ num);
            i += 1;
        }
        return result;
    }
}